web-common/src/features/feature-flags.ts

```diff
@@ -6,3 +7,14 @@ export type FeatureFlags = {
 export const featureFlags = writable<FeatureFlags>({
   readOnly: undefined,
 });
+
+export function retainFeaturesFlags(navigation: BeforeNavigate) {
+  if (!navigation.from.url.searchParams.has("features")) {
+    return;
+  }
+
+  navigation.to.url.searchParams.set(
+    "features",
+    navigation.from.url.searchParams.get("features")
```

Comment:
In your code you use the search param `features`, but in your tests you are using the param `feature` (no 's').
Sounds like we need to pick with one and stick with it. Doesn't really matter to me but I'm used to it being just `feature` in other systems I've used